benchmark.py

```diff
@@ -0,0 +1,61 @@
+#! /usr/bin/env python2.7
+
+from __future__ import print_function
+from grammars.grammars import java
+from treemanager import TreeManager
+from incparser.astree import BOS, EOS
+from time import time
+import sys, os
+
+def run(filename):
+    # Load grammar
+    parser, lexer = java.load()
+    treemanager = TreeManager()
+    treemanager.add_parser(parser, lexer, "")
+
+    # Load file
+    print("Running", filename)
+    with open(filename, "r") as f:
+        treemanager.import_file(f.read())
+
+    if parser.last_status is False:
+        print("Couldn't parse file. Skip.")
+        return
+
+    filesize = os.path.getsize(filename)
+
+    timings = []
+
+    # Insert `1+` after every `=`
+    node = treemanager.get_bos()
+    while type(node) is not EOS:
+        if node.symbol.name == "=":
+            node.insert("1+", len(node.symbol.name))
+            node.mark_changed()
+            treemanager.relex(node)
+            treemanager.post_keypress("")
+            treemanager.save_current_version()
+            parser.prev_version = treemanager.version
+            parser.reference_version = treemanager.reference_version
+            start = time()
+            parser.inc_parse()
+            end = time()
+            parser.top_down_reuse()
+            treemanager.save_current_version(postparse=True) # save post parse tree
+            if parser.last_status == True:
+                treemanager.reference_version = treemanager.version
+            TreeManager.version = treemanager.version
+
+            if parser.last_status is False:
+                break
+            timings.append(str(end-start))
+        node = node.next_term
+
+    if timings:
+        with open("results.csv", "a+") as f:
```

Comment:
Are you sure you want append here? For experiments, I tend to favour a "fail hard" approach, which here would mean checking if `results.csv` exists, and failing if it does. If it doesn't exist, then `"w"` would be fine here.

Reply:
But then I'd have to create an empty `results.csv` before each run. I guess I could add `touch results.csv` to `build.sh`

Reply:
If you do `open("results.csv", "w")` you'll be fine, right?

Reply:
No, because `benchmark.py` is run for every file in the library, so I need to append the results after every run.

Reply:
Oh, I see. OK, then I guess you'll have to do the "file exists: it's your job to delete it" stuff in `build.sh`.

Reply:
Yep, already done that. Commit coming soon.